app/scripts/controllers/metamask-notifications/metamask-notifications.ts

```diff
@@ -836,6 +1113,52 @@ export class MetamaskNotificationsController extends BaseController<
       s.metamaskNotificationsReadList = [
         ...new Set([...currentReadList, ...newReadIds]),
       ];
+
+      s.metamaskNotificationsList = s.metamaskNotificationsList.map(
+        (notification: Notification) => {
+          if (newReadIds.includes(notification.id)) {
+            return { ...notification, isRead: true };
+          }
+          return notification;
+        },
+      );
     });
   }
+
+  /**
+   * Updates the list of MetaMask notifications by adding a new notification at the beginning of the list.
+   * This method ensures that the most recent notification is displayed first in the UI.
+   *
+   * @param notification - The new notification object to be added to the list.
+   * @returns A promise that resolves when the notification list has been successfully updated.
+   */
+  public async updateMetamaskNotificationsList(
+    notification: Notification,
+  ): Promise<void> {
+    const isNotUndefined = <T>(t?: T): t is T => Boolean(t);
+    const processAndFilterSingle = (
+      n: FeatureAnnouncementRawNotification | OnChainRawNotification,
+    ) => {
+      try {
+        const processedNotification = processNotification(n);
+        if (isNotUndefined(processedNotification)) {
+          return processedNotification;
+        }
+      } catch {
+        return undefined;
+      }
+      return undefined;
+    };
+
+    const processedNotification = processAndFilterSingle(notification);
+
+    if (processedNotification) {
+      this.update((s) => {
+        s.metamaskNotificationsList = [
+          notification,
+          ...s.metamaskNotificationsList,
+        ];
+      });
+    }
```

Comment:
And I guess this is for updating our "cache" with the push notification.

OK, but add some deduping just in case.

Reply:
Dedupe by id - future ticket.

Reply:
Ticket: https://consensyssoftware.atlassian.net/browse/NOTIFY-656

Reply:
Fixed.